pennylane/ops/op_math/composite.py

```diff
@@ -152,7 +149,10 @@ def is_hermitian(self):
     # pylint: disable=arguments-renamed, invalid-overridden-method
     @property
     def has_matrix(self):
-        return all(op.has_matrix or isinstance(op, qml.Hamiltonian) for op in self)
+        return all(
+            op.has_matrix or isinstance(op, (qml.ops.Hamiltonian, qml.ops.LinearCombination))
```

Comment:
```suggestion
            op.has_matrix or isinstance(op, qml.ops.Hamiltonian)
```
Again not needed since `LinearCombination` should define a matrix.